src/java/org/apache/cassandra/config/Config.java

```diff
@@ -88,6 +88,8 @@ public class Config
 
     public long native_transport_idle_timeout_in_ms = 0L;
 
+    public volatile Long internode_connect_timeout_in_ms = 1000L;
```

Comment:
So we intentionally want to lower this value for steady state instead of just overriding it for migration? Have we seen this causing issues outside of migrations?

Reply:
Yes I'd like to have this just be a default config. We already run a significant portion of clusters with read/write timeouts of 1s, so I'm not worried about inserting this here since I believe socket connection should be fairly independent of cassandra query-processing-time

If there is _any_ "longer"-lived connectivity problem between nodes (or you do a regular restart + another node doesn't yet see the restarting one as down) I expect this to be beneficial.

Reply:
Do we have any concerns about weird onprem setups where networking might be slower and the SSL handshake might take a little longer?

Reply:
I don't think so -- we run 1s timeout clusters onprem without issue in a few places already (and the oss default is already 2s)